here-naksha-lib-core/src/main/java/com/here/naksha/lib/core/AbstractTask.java

```diff
@@ -393,24 +392,40 @@ protected final void unlock() {
    * @throws RuntimeException      If adding the task to the thread pool failed for an unknown error.
    */
   public @NotNull Future<@NotNull RESULT> start() {
-    final long LIMIT = AbstractTask.limit.get();
+    final long LIMIT = requestLimitManager.getInstanceLevelLimit();
     lockAndRequireNew();
     try {
+      String actorId = context.getActor();
+      final long ACTOR_LIMIT = requestLimitManager.getActorLevelLimit(context);
+      incActorLevelUsage(actorId, ACTOR_LIMIT);
       do {
         final long threadCount = AbstractTask.threadCount.get();
         assert threadCount >= 0L;
         if (!internal && threadCount >= LIMIT) {
-          throw new TooManyTasks();
+          log.info(
+              "NAKSHA_ERR_REQ_LIMIT_4_INSTANCE - [Request Limit breached for Instance => appId,author,actor,limit,crtValue] - ReqLimitForInstance {} {} {} {} {}",
+              actorId == null ? "" : context.getAppId(),
```

Comment:
There are unit tests for which actor value is null, If that falls into this condition context.getAppId() will throw error. Thats why added ternary operator. Let me know what appId value we need to provide in this case.
`actorId == null ? "" : context.getAppId()`

Reply:
There is one unit test which tests the tooManyException by exceeding the limits, In that case i am seeing the AppId is null. If AppId is null then it throws Unauthorized exception. To handle such kind of issues i added this condition.

Reply:
I am not sure if we are talking about same thing on this comment. My comment is related to logging (to log appId's true value) and not about TooManyException.

Reply:
I was trying to explain with an example unit test case. I am seeing appId is null for one of the unit test, in that case while logging it is throwing an exception. To handle that i added null check.

Reply:
Fixing It.